processing/src/main/java/io/druid/segment/data/IntermediateLongSupplierSerializer.java

```diff
@@ -42,6 +44,7 @@
   private final ByteOrder order;
   private final CompressedObjectStrategy.CompressionStrategy compression;
   private CountingOutputStream tempOut = null;
+  private final ByteBuffer helperBuffer = ByteBuffer.allocate(Longs.BYTES);
```

Comment:
This class is thread-unsafe not only because of helperBuffer, `CountingOutputStream tempOut` and `BiMap uniqueValues` are also thread-unsafe. Added a Javadoc note.